### PR TITLE
Elegoo Neptune Boards

### PR DESCRIPTION
**Pull request by mlee12382**
### Description
Added new board definitions and pins and updated .ini files for platformio for the Elegoo Neptune series of printers.

### Requirements
N/A

### Benefits
Adds boards to make custom compiling firmware for the Elegoo Neptune series of printers easier, duplicates and renames the 2 boards that come in the Neptune 2 printers to make it more straightforward and adds the missing board files for the Neptune X and Neptune 3

### Configurations
N/A

### Related Issues
N/A

### CODE REVIEW

**Comment by EvilGremlin** [comment]
Redundant, ZNP_ROBIN_NANO is exactly same as MKS_ROBIN_NANO

**Comment by mlee12382** [comment]
> Redundant, ZNP_ROBIN_NANO is exactly same as MKS_ROBIN_NANO

The board is a clone of the MKS but it is labeled ZNP on the control board, the redundancy is to avoid confusion for people wanting to build their own custom firmware.

**Comment by EvilGremlin** [comment]
MKS Robin (nano) E3 also exist already. 
Which leaves quiestion: is Neptune X board unique or also rebadge of something else?

To avoid confusion simple name alias is more than enough, no need to make a copy of pin files
#elif MB(ZNP_ROBIN_NANO)
  #include "stm32f1/pins_MKS_ROBIN_NANO.h"
like that

**Comment by mlee12382** [comment]
> MKS Robin (nano) E3 also exist already. Which leaves quiestion: is Neptune X board unique or also rebadge of something else?
> 
> To avoid confusion simple name alias is more than enough, no need to make a copy of pin files #elif MB(ZNP_ROBIN_NANO) #include "stm32f1/pins_MKS_ROBIN_NANO.h" like that

Neptune X is unique as is Neptune 3, I think that's how I did the znp robin boards was just as aliases in boards.h

File: Marlin/src/pins/stm32f4/pins_NEPTUNE_3.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+// Avoid conflict with TIMER_TONE
+#include "env_validate.h"
+//
+#define ALLOW_STM32F4
+
+// Release PB4 (Z_DIR_PIN) from JTAG NRST role
+//
+// #define DISABLE_DEBUG
+
+#define BOARD_INFO_NAME "MKS Robin Nano E3"
```
Comment (EvilGremlin) [inline Marlin/src/pins/stm32f4/pins_NEPTUNE_3.h]: Yep, this is mistake, not nano e3 indeed

Comment (mlee12382) [reply Marlin/src/pins/stm32f4/pins_NEPTUNE_3.h]: That was provided to me by elegoo, they based it on the nano e3 but it is custom

Comment (mlee12382) [reply Marlin/src/pins/stm32f4/pins_NEPTUNE_3.h]: I just updated the name to fix the confusion

File: ini/stm32f1.ini [hunk 2]
```diff
@@ -233,6 +233,24 @@ build_unflags               = ${stm32_variant.build_unflags}
 debug_tool                  = jlink
 upload_protocol             = jlink
 
+#
+# ZNP Robin Nano V1.2
+#
+[env:znp_robin_nano35]
```
Comment (EvilGremlin) [inline ini/stm32f1.ini:239]: also no need for full copy, just set 
`extends = mks_robin_nano35
board_build.encrypt_mks = elegoo.bin`
also can be `#define FIRMWARE_BIN elegoo.bin` in example config

Comment (mlee12382) [reply ini/stm32f1.ini:239]: *facepalm* idk why I didn't think of that.. lol

**Comment by EvilGremlin** [comment]
Do you have board photos? Google is silent about new ones

**Comment by mlee12382** [comment]
Only these from the user guides for the Neptune X and Neptune 3, do you need the Neptune 2 boards also?
NEPTUNE X
NEPTUNE 3

**Comment by EvilGremlin** [comment]
Neptune X is robin nano 1.3...
defines USES_DIAG_JUMPERS and CS pins etc but photo says there are none

**Comment by mlee12382** [comment]
> Neptune X is robin nano 1.3...

No the pins don't match anything currently in marlin.
Neptune-X

**Comment by mlee12382** [comment]
It's close but not 100% match

**Comment by EvilGremlin** [comment]
Yep, i see... a bit of a mess. Do you have actual hardware to verify pinouts? Bacause this neptune X v1.0 photo clearly don't match pins.

**Comment by mlee12382** [comment]
> Yep, i see... a bit of a mess. Do you have actual hardware to verify pinouts? Bacause this neptune X v1.0 photo clearly don't match pins.

No unfortunately not :( I am trying to help out the community cause we've been lacking firmware other than stock for the X and the 3 is brand new. I may need to do a bit more research it seems.

**Comment by mlee12382** [comment]
> Neptune X is robin nano 1.3... defines USES_DIAG_JUMPERS and CS pins etc but photo says there are none

I copied the nano_v3 and modified the pins since that is what elegoo said was the board they cloned and modified, I can remove that definition if it's not relevant.

**Comment by EvilGremlin** [comment]
Nah, let's leave it as is for now. Maybe someone with hardware will show up to verify.

**Comment by mlee12382** [comment]
> Nah, let's leave it as is for now. Maybe someone with hardware will show up to verify.

I just noticed that the cs pins interfere with the fsmc pins, not sure why I didn't catch that before, will fix it real quick.

**Comment by thinkyhead** [comment]
Does this need additional verification or should it be merged in its current state?

**Comment by EvilGremlin** [comment]
I print with this conf for a week (no bltouch though, might have missed something there in revamp)

**Comment by EvilGremlin** [comment]
oops, wrong thread :)

**Comment by mlee12382** [comment]
> Does this need additional verification or should it be merged in its current state?
 

I need to do more work on this before it's ready for merging unfortunately  :(

**Comment by NotLordCalvert** [comment]
I realize this is closed, but would a picture of the board help here? If not, what do you need @mlee12382?

**Comment by mlee12382** [comment]
> I realize this is closed, but would a picture of the board help here? If not, what do you need @mlee12382?

For the Neptune 3 I need to figure out why it is having trouble compiling and why the SanityCheck is convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. The "official" N3 firmware bypasses that line of the SanityCheck instead of fixing the underlying issue which won't fly with the actual Marlin git.

**Comment by ellensp** [comment]
"convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. "
This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files

**Comment by mlee12382** [comment]
> "convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. "
> This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files 

I will update my marlin source and try again, I did update fairly recently though and it was still an issue,  but will try again just in case.

**Comment by mlee12382** [comment]
> "convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. " This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files

Unfortunately even updating to the current 2.1.x bugfix files I am still getting the same error. :(

**Pull request closed without merging**